projects/tutorial_102/src/main.c

```diff
@@ -0,0 +1,44 @@
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct Counters {
+        uint8_t counter_a;
+        uint8_t counter_b;
+    } Counters;
+
+static void prv_timer_callback(SoftTimerId timer_id, void *context) {
+    Counters *storage = context;
+    storage->counter_a++;
+
+    LOG_DEBUG("Counter A: %d\n", storage->counter_a);
+    
+    if (storage->counter_a == (storage->counter_b)*2 + 2) {
+        storage->counter_b++;
+        LOG_DEBUG("Counter B: %d\n", storage->counter_b);
+    }
+
+    soft_timer_start_millis(500,
+                            prv_timer_callback,
+                            storage,
+                            NULL);
```

Comment:
Please use a `#define`'d constant, nicely named and including the unit (ms), for the 500 instead of using 500 here and below. It makes it easier to read and change. Also it's not really necessary to break the function call over multiple lines, a `make format` should take care of it.